### PR TITLE
refactor: Add ability to lockdown NTSC [DET-8276, DET-7377]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

- Adds a authz function ```CanAccessNTSC``` which is called to allow for enabling allowing non-admin users to view their own tasks.

- Adds a dev master config option ```_strict_ntsc_enabled``` allowing for testing of a strict task world

- Checks authz for getting experiments in tensorboard creation

- Adds an authz function for ```GetActiveTasksCount```

## Test Plan

- E2e test covering strict NTSC

- Go integration test for authz for tensorboard creation

- Go integration test for ```GetActiveTasksCount``` authz

## Commentary (optional)

The second ticket is for tests around ```/proxy``` and I think this covers ensuring the proxy is authenticated. 

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: master/internal/user/service.go [hunk 1]
```diff
@@ -203,17 +204,28 @@ func (s *Service) ProcessProxyAuthentication(c echo.Context) (done bool, err err
...
-	default:
+	} else if err != nil {
 		return true, err
 	}
+	if !user.Active {
+		return true, redirectToLogin(c)
+	}
+
+	taskID := c.Param("service")
+	ownerID, err := db.GetCommandOwnerID(c.Request().Context(), model.TaskID(taskID))
```
Comment (NicholasBlaskey) [inline master/internal/user/service.go:218]: Is it correct to say this proxy only auth only used by notebooks and tensorboards now? 

I tried to dig through the code but couldn't really tell if I was missing other use cases of this proxy auth.

File: .circleci/devcluster/strict-ntsc.devcluster.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            rbac_ui_enabled: true
+            _strict_ntsc_enabled: true
+            
+  - custom:
+      name: proxy
+      cmd: ["socat", "-d", "-d", "TCP-LISTEN:8081,reuseaddr,fork", "TCP:localhost:8080"]
+      post:
+        - conncheck:
+            port: 8081
+
```
Comment (ioga) [inline .circleci/devcluster/strict-ntsc.devcluster.yaml:35]: if you aren't breaking the connection between agent and master in the test I'd say you can safely omit this proxy step (remove it and use port `8080` in agent configs)

File: e2e_tests/tests/command/test_strict_ntsc.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    master_url = conf.make_master_url()
+    authentication.cli_auth = authentication.Authentication(
+        master_url, creds.username, creds.password, try_reauth=True
+    )
+    sess = api.Session(master_url, None, None, None)
+
+    resp = bindings.get_GetShells(sess)
+    shell_ids = [shell.id for shell in resp.shells or []]
+    if can_access:
+        assert shell_id in shell_id
```
Comment (ioga) [inline e2e_tests/tests/command/test_strict_ntsc.py:24]: I suspect
```suggestion
        assert shell_id in shell_ids
```

File: e2e_tests/tests/command/test_strict_ntsc.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    assert shell_id not in shell_ids
+    with pytest.raises(NotFoundException):
+        bindings.get_GetShell(sess, shellId=shell_id)
+    with pytest.raises(NotFoundException):
+        bindings.post_KillShell(sess, shellId=shell_id)
+    with pytest.raises(NotFoundException):
+        req = bindings.v1SetShellPriorityRequest(shellId=shell_id, priority=50)
+        bindings.post_SetShellPriority(sess, shellId=shell_id, body=req)
+    with pytest.raises(NotFoundException):
+        api.get(master_url, f"shells/{shell_id}/events", auth=authentication.cli_auth)
```
Comment (ioga) [inline e2e_tests/tests/command/test_strict_ntsc.py:43]: do we test if "does this shell id show up in the list of shells" anywhere?

File: master/internal/user/service.go (hunk 1 above)
Comment (ioga) [reply master/internal/user/service.go:218]: hmm yes, also shells. 

when we'll be shipping "heroes of determined 3" / generic proxies, this proxy will be used by any task type, but that's future work.

**Review by ioga — APPROVED** [review]
lgtm

File: e2e_tests/tests/command/test_strict_ntsc.py (hunk 4 above)
Comment (NicholasBlaskey) [reply e2e_tests/tests/command/test_strict_ntsc.py:43]: for the negative case we test on line 34  ```assert shell_id not in shell_id``` 

and for the positive case we test on line 24

**Pull request merged**